contracts/escrows/CountdownGriefingEscrow.sol

```diff
@@ -0,0 +1,357 @@
+            _data.seller = msg.sender;
+        }
+        // restrict state machine
+        require(!isFinalized() && !isCancelled(), "only before finalize or cancel");
+
+        // Add the stake amount
+        if (uint256(_data.stakeAmount) != uint256(0))
+            Staking._addStake(_data.seller, msg.sender, uint256(0), uint256(_data.stakeAmount));
+
+        // If payment is deposited, finalize the escrow
+        if (isPaymentDeposited())
+            _data.status = EscrowStatus.isDeposited;
+            finalize();
+
+        // emit event
+        emit StakeDeposited(_data.seller, uint256(_data.stakeAmount));
+    }
+
+    /// @notice Deposit Payment
+    ///          - if buyer not already set, make msg.sender the buyer
+    ///          - if seller already deposited the stake, start the finalization countdown
+    /// @dev Access Control: buyer OR operator
+    ///      State Machine: before finalize() OR before cancel()
+    function depositPayment() public {
+        // restrict access control
+        // set msg.sender as buyer if not already set
+        if (!isBuyer(msg.sender) && !Operated.isActiveOperator(msg.sender)) {
+            require(_data.buyer == address(0), "only buyer or active operator");
+            _data.buyer = msg.sender;
+        }
+        // restrict state machine
+        require(!isFinalized() && !isCancelled(), "only before finalize or cancel");
+
+        // Add the payment as a stake
+        if (uint256(_data.paymentAmount) != uint256(0))
+            Staking._addStake(_data.buyer, msg.sender, uint256(0), uint256(_data.paymentAmount));
+
+        // If stake is deposited, start countdown for seller to finalize
+        if (isStakeDeposited())
+            _data.status = EscrowStatus.isDeposited;
+            Countdown._start();
+
+        // emit event
+        emit PaymentDeposited(_data.buyer, uint256(_data.paymentAmount));
+    }
+
+    /// @notice Finalize escrow and execute completion script
+    ///          - create the agreement
+    ///          - transfer stake and payment to agreement
+    ///          - start agreement countdown
+    ///          - disable agreement operator
+    /// @dev Access Control: seller OR operator
+    ///      State Machine: after depositStake() AND after depositPayment()
+    function finalize() public {
+        // restrict access control
+        require(isSeller(msg.sender) || Operated.isActiveOperator(msg.sender), "only seller or active operator");
+        // restrict state machine
+        require(isDeposited(), "only after deposit");
+
+        // create the agreement
+
+        address agreement;
+        {
+            // get the agreement factory address
+
+            address escrowFactory = Template.getFactory();
+            address escrowRegistry = iFactory(escrowFactory).getInstanceRegistry();
+            address agreementFactory = abi.decode(iRegistry(escrowRegistry).getFactoryData(escrowFactory), (address));
```

Comment:
I'm not a big fan of `getFactoryData`. It feels like you're adding this weird undocumented interface for what type `extraData` should be, instead of just making it a real method with an actual interface. If you go down this route, it probably makes the most sense to have it as a member of `CountdownGriefingEscrow_Factory`.

Reply:
Keep as is for now, but agree makes sense to add to factory